sktime/forecasting/base/adapters/_neuralforecast.py

```diff
@@ -134,16 +136,49 @@ def algorithm_parameters(self: "_NeuralForecastAdapter") -> dict:
         - custom model name (``alias``) - used from ``algorithm_name``
         """
 
+    def _get_valid_parameters(self: "_NeuralForecastAdapter") -> dict:
+        """Get valid parameters for the underlying NeuralForecast algorithm class.
+
+        Returns
+        -------
+        dict
+            valid list of arguments for the underlying algorithm class
+        """
+        from pytorch_lightning import Trainer
+
+        model_class = self.algorithm_class
+        trainer_params = list(signature(Trainer.__init__).parameters.keys())
+        valid_parameters = list(signature(model_class.__init__).parameters.keys())
+        valid_parameters += trainer_params
+
+        sktime_parameters = self.algorithm_parameters
+
+        for sktime_param in list(sktime_parameters.keys()):
+            if sktime_param not in valid_parameters:
+                if sktime_parameters[sktime_param] is not None:
+                    warn(
+                        f"Keyword argument '{sktime_param}' will be omitted as it is"
+                        f" not found in the __init__ method "
+                        f"from {self.algorithm_class}. "
+                        f"Check your neuralforecast version"
```

Comment:
Question:  will it make sense to raise different warnings for discrepancy with neuralforecast or with pytorch lightning?

Reply:
We have issues with params of neuralforecast, so why raise warnings for pytorch lightning ?? I'm unable to understand what are you trying to say here

Reply:
Suppose neuralforecast adds upper  bound for pytorch lightning to version X, where X changes over time. And suppose Y is the latest version of pytorch lightning, which also vary over time.

If Y has few parameters that X lacks, and user sees them from the website and passes them to trainer arguments, it will fail on sktime main as those are unsupported. If I understand correctly, your changes will be able to capture and drop those extra ones, but the warning will inform users about neuralforecast version limitation. But these warnings will originate due to pytorch lightning version discrepancies, not neuralforecast. 

Let me know if it makes sense.

Reply:
I don't expect a class like `Trainer` to add a new param in further versions or update. Still if you think adding a warning for params not supported due pytorch_ligthning version I will add that. But this won't happen most probably. Let me know what do you think about it ??